packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -115,19 +122,28 @@ export class PendingStateManager implements IDisposable {
     constructor(
         private readonly containerRuntime: ContainerRuntime,
         private readonly applyStashedOp: (type, content) => Promise<unknown>,
-        initialState: IPendingLocalState | undefined,
+        initialFlushMode: FlushMode,
+        initialLocalState: IPendingLocalState | undefined,
     ) {
-        this.initialStates = new Deque<IPendingState>(initialState?.pendingStates ?? []);
+        this.initialStates = new Deque<IPendingState>(initialLocalState?.pendingStates ?? []);
 
-        if (initialState) {
-            if (initialState?.clientId) {
-                this.previousClientIds.add(initialState.clientId);
+        if (initialLocalState) {
+            if (initialLocalState?.clientId) {
+                this.previousClientIds.add(initialLocalState.clientId);
             }
             // get stashed op count and client sequence number of first op
-            const messages = initialState.pendingStates
+            const messages = initialLocalState.pendingStates
                 .filter((state) => state.type === "message") as IPendingMessage[];
             this.firstStashedCSN = messages[0].clientSequenceNumber;
         }
+
+        /**
+         * The initial flush mode is be default the first acked one because we want to start with this if the very first
```

Comment:
I think this needs to be rephrased

Reply:
I renamed the `lastAckedFlushMode` to `flushModeForNextMessage` for clarity and rephrased the comments. Please take a look and let me know if it makes more sense now.